app/controllers/DeclarationController.scala

```diff
@@ -57,10 +57,9 @@ class DeclarationController @Inject()(actions: Actions, client: CustomsDeclarati
     }
   }
 
-  def displaySubmitConfirmation(conversationId: String): Action[AnyContent] = (actions.switch(Feature.submit) andThen actions.auth).async { implicit req =>
-    submissionRepository.getByConversationId(conversationId).map {
-      case Some(submission) => Ok(views.html.submit_confirmation(submission))
-      case None => NotFound(views.html.error_template("Submission Not Found", "Submission Not Found", "We're sorry but we couldn't find that submission.")) // TODO throw specific ApplicationException type to be handled via ErrorHandler
+  def displaySubmitConfirmation(conversationId: String): Action[AnyContent] = (actions.switch(Feature.submit) andThen actions.auth).async {
```

Comment:
As you have to wrap every outcome of this action in a `Future.successful` you can just remove the `.async`
```suggestion
  def displaySubmitConfirmation(conversationId: String): Action[AnyContent] = (actions.switch(Feature.submit) andThen actions.auth) {
```